nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/repository/Repository.java

```diff
@@ -84,6 +93,14 @@ public boolean isDefault() {
         return Boolean.TRUE.equals(isDefault);
     }
 
+    /** @since 11.2 */
+    public boolean isHeadless() {
+        if (isDefault()) {
+            return Boolean.TRUE.equals(headless);
+        }
+        return !Boolean.FALSE.equals(headless);
```

Comment:
Should we understand this as "if the repository is not the default one and `headless` is not filled in (`null`), then it's considered as headless"?
In this case, why is it different from master, see https://github.com/nuxeo/nuxeo/commit/e018af03de77189b05dc32357260f202b47e49d1#diff-e73e25622874c080fe10b7cbcb291ea8R93?
Might deserve a comment.

Reply:
yes, this is different than in master because we want all repos to be headless by default unless it's the default one (if this is not explictly set to be headless).
this is so that a repository picker doesn't show up all of sudden after installing a HF.

Reply:
OK thanks.
Again, I think this explanation would be interesting as a comment in the code.

Reply:
+1 on adding a comment explaining that this is for backward compat

Reply:
👍 added a comment